cmd/prometheus/config.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/prometheus/log"
+	"github.com/prometheus/prometheus/notification"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/storage/local"
+	"github.com/prometheus/prometheus/storage/local/index"
+	"github.com/prometheus/prometheus/web"
+)
+
+// cfg contains immutable configuration parameters for a running Prometheus
+// server. It is populated by its flag set.
+var cfg = struct {
+	fs *flag.FlagSet
+
+	printVersion bool
+	configFile   string
+
+	storage      local.MemorySeriesStorageOptions
+	notification notification.NotificationHandlerOptions
+	queryEngine  promql.EngineOptions
+	web          web.Options
+
+	// Remote storage.
+	remoteStorageTimeout  time.Duration
+	influxURL             string
```

Comment:
Let's stick to `influx` -> `influxdb`? Feels weird to abbreviate the product name like that...